desktop/core/src/desktop/js/apps/newimporter/FilePreviewTab/FilePreviewTab.test.tsx

```diff
@@ -125,8 +146,12 @@ describe('FilePreviewTab', () => {
       />
     );
 
+    const editColumnsButton = screen.getByText('Edit Columns');
+
+    await userEvent.click(editColumnsButton);
+
     await waitFor(() => {
-      expect(screen.getByText('Configure source')).toBeInTheDocument();
+      expect(screen.getByRole('dialog')).toBeInTheDocument();
```

Comment:
The problem with `toBeInTheDocument` is that it only checks if the element is in the DOM. Your EditColumnsModal is always in the DOM, sometimes hidden, sometimes showing.
We should test that EditColumnsModal is hidden before the the click and then that it is visible after the click.